pkg/trace/agent/agent_test.go

```diff
@@ -581,3 +586,165 @@ func formatTrace(t pb.Trace) pb.Trace {
 	}
 	return t
 }
+
+func BenchmarkThroughput(b *testing.B) {
+	if _, ok := os.LookupEnv("DD_TRACE_TEST_PRIVATE_DATA"); !ok {
+		b.SkipNow()
+	}
+
+	ddlog.SetupDatadogLogger(seelog.Disabled, "") // disable logging
+
+	b.Run("39KB/10", benchThroughput("src/github.com/DataDog/trace-agent-test/payloads/10traces.msgp"))
+	b.Run("1.1MB/100", benchThroughput("src/github.com/DataDog/trace-agent-test/payloads/100traces.msgp"))
+	b.Run("1.1MB/1951", benchThroughput("src/github.com/DataDog/trace-agent-test/payloads/26.msgp"))
+	b.Run("2.2MB/4161", benchThroughput("src/github.com/DataDog/trace-agent-test/payloads/10.msgp"))
+	b.Run("6.6MB/500", benchThroughput("src/github.com/DataDog/trace-agent-test/payloads/500traces.msgp"))
+}
+
+func benchThroughput(file string) func(*testing.B) {
+	return func(b *testing.B) {
+		data, count, err := tracesFromFile(file)
+		if err != nil {
+			b.Fatal(err)
+		}
+		cfg := config.New()
+		cfg.Endpoints[0].APIKey = "irrelevant"
+
+		ctx, cancelFunc := context.WithCancel(context.Background())
+		http.DefaultServeMux = &http.ServeMux{}
+		agnt := NewAgent(ctx, cfg)
```

Comment:
nit: `agent`

Reply:
I guess it works here. I avoid it because it's the name of a package 😄